src/ArrayInterface.jl

```diff
@@ -445,13 +445,9 @@ Is strides(::T) defined? It is assumed that types returning `true` also return a
 pointer on `pointer(::T)`.
 """
 defines_strides(x) = defines_strides(typeof(x))
-function defines_strides(::Type{T}) where {T}
-    if parent_type(T) <: T
-        return false
-    else
-        return defines_strides(parent_type(T))
-    end
-end
```

Comment:
ouch.

Reply:
Maybe we should file a Julia issue on the branch no longer inferring?

Reply:
Looks similar to https://github.com/JuliaLang/julia/issues/43368.